src/perm/mod.rs

```diff
@@ -199,6 +200,8 @@ impl PartialEq for Permutation {
     }
 }
 
+impl Eq for Permutation {}
```

Comment:
Since it is a marker trait, we probably can implement this with a derive which would be slightly cleaner (personal preference)

Reply:
Changed now